src/test/java/com/cloudbees/jenkins/plugins/sshagent/SSHAgentBase.java

```diff
@@ -71,10 +68,9 @@ public class SSHAgentBase {
             "-----END OPENSSH PRIVATE KEY-----";
 
     protected void startMockSSHServer() throws Exception {
-        File hostKey = new File(System.getProperty("java.io.tmpdir") + "/key.ser");
-        hostKey.delete(); // do not carry from test to test
+        File hostKey = temp.newFile();
         sshd = SshServer.setUpDefaultServer();
-        sshd.setPort(getValidPort());
+        sshd.setPort(0);
```

Comment:
get the SSH server to start on a random port - so that it is guaranteed to be available.